src/components/Snapshots/SnapshotsListView/SnapshotsColumnManagement.tsx

```diff
@@ -0,0 +1,114 @@
+import React from 'react';
+import {
+  Modal,
+  ModalVariant,
+  Button,
+  Form,
+  FormGroup,
+  Checkbox,
+  Text,
+  TextContent,
+  TextVariants,
+  Flex,
+  FlexItem,
+} from '@patternfly/react-core';
+import { SnapshotColumnKey, snapshotColumns, defaultVisibleColumns } from './SnapshotsListHeader';
+
+type SnapshotsColumnManagementProps = {
+  isOpen: boolean;
+  onClose: () => void;
+  visibleColumns: Set<SnapshotColumnKey>;
+  onVisibleColumnsChange: (columns: Set<SnapshotColumnKey>) => void;
+};
+
+const SnapshotsColumnManagement: React.FC<SnapshotsColumnManagementProps> = ({
+  isOpen,
+  onClose,
+  visibleColumns,
+  onVisibleColumnsChange,
+}) => {
+  const [localVisibleColumns, setLocalVisibleColumns] =
+    React.useState<Set<SnapshotColumnKey>>(visibleColumns);
+
+  React.useEffect(() => {
+    if (isOpen) {
+      setLocalVisibleColumns(new Set(visibleColumns));
+    }
+  }, [isOpen, visibleColumns]);
+
+  const handleColumnToggle = (columnKey: SnapshotColumnKey) => {
+    const newColumns = new Set(localVisibleColumns);
+    if (newColumns.has(columnKey)) {
+      // Don't allow hiding the name column or kebab column
+      if (columnKey !== 'name' && columnKey !== 'kebab') {
```

Comment:
maybe we could add these keys into an array (e.g. something like `const NON_HIDABLE_COLUMNS = ['name', 'kebab'];`) and use it in this file where there's a logic related to it. So here we could update to:

```ts
if (!NON_HIDABLE_COLUMNS.includes(columnKey)) {
```

Reply:
Done